ttnn/cpp/ttnn/operations/data_movement/copy/device/copy_device_operation.cpp

```diff
@@ -32,9 +32,9 @@ void CopyDeviceOperation::validate_with_output_tensors(const std::vector<Tensor>
     }
     DataType output_dtype = this->output_dtype;
     if(!output_tensors.empty() && output_tensors.at(0).has_value()){
-        const auto output_shape_required = this->compute_output_shapes(input_tensors);
         const auto& out_tensor = output_tensors.at(0).value();
-        TT_FATAL(out_tensor.get_legacy_shape() == output_shape_required.at(0), "The input tensors need a shape of {}, however the output tensor is only {}", output_shape_required,  out_tensor.get_legacy_shape());
+        const auto output_shape_required = input_tensors.size() == 2 ? input_tensors[1].get_legacy_shape() : input_tensor_a.legacy_shape();
+        TT_FATAL(out_tensor.get_legacy_shape() == output_shape_required, "The input tensors need a shape of {}, however the output tensor is only {}", output_shape_required,  out_tensor.get_legacy_shape());
```

Comment:
Would it better to unpack this into two checks, one for logical shape and one for padded? This will help remove dependency on `==` operator of `LegacyShape`.